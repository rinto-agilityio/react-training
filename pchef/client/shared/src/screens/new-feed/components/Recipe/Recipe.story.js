// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'

// Helpers
import { recipes } from '../../../../mocks'

// Components
import Recipe from '.'

// Helper
import { wInfo } from '../../../../../.storybook/utils';

storiesOf('New feed recipe', module)
  .addDecorator(wInfo())
  .add('Medium', () => (
    <Recipe size="medium" recipe={recipes[0]} favoriteRecipe={['idtest']} />
  ))
  .add('Large', () => (
    <Recipe size="large" recipe={recipes[0]} favoriteRecipe={['idtest']} />
  ))
