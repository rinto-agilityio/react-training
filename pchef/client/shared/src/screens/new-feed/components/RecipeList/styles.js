// Lib
import { StyleSheet } from 'react-native'

// Themes
import { METRICS, COLORS } from '../../../../themes'

// Create style for Header
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  primaryContainer: {
    paddingTop: METRICS.extraLargePadding,
  },
  secondaryContainer: {
    paddingTop: METRICS.extraLargePadding + METRICS.largePadding,
    maxWidth: METRICS.extraLargeScreen,
    width: '100%',
    margin: 'auto',
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'flex-start',
  },
  indicator: {
    marginTop: METRICS.largeMargin * 10,
  },
  tabWrap: {
    display: 'flex',
    flexDirection: 'row',
    borderBottomWidth: METRICS.smallBorderWidth,
    borderColor: COLORS.baseGray,
    marginTop: METRICS.largeMargin,
    marginBottom: METRICS.largeMargin,
  },
  tab: {
    paddingTop: METRICS.mediumPadding,
    paddingRight: METRICS.largePadding,
    paddingBottom: METRICS.mediumPadding,
    paddingLeft: METRICS.largePadding,
    color: COLORS.white,
    textTransform: 'uppercase',
    borderWidth: METRICS.smallBorderWidth,
    borderTopLeftRadius: METRICS.mediumBorderRadius,
    borderTopRightRadius: METRICS.mediumBorderRadius,
  },
  tabViewActive: {
    backgroundColor: COLORS.red,
  },
  tabView: {
    backgroundColor: COLORS.baseGray,
  },
})

export default styles
