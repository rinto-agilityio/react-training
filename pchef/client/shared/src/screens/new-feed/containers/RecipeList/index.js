import { compose } from 'react-apollo'

// GraphQL
import { userToggleRecipe, userToggleVote } from '../../../../graphql/recipe'
import RecipeList from '../../components/RecipeList'

export default compose(
  userToggleVote,
  userToggleRecipe,
)(RecipeList)
