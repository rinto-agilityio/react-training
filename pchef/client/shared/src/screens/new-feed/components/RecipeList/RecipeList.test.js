// Libs
import { act } from 'react-dom/test-utils'

// Components
import { Text } from 'react-native'
import RecipeList from '.'
import Loading from '../../../../components/Loading'
import Recipe from '../Recipe'

// Mocks
import { recipes } from '../../../../mocks'
import { NO_RECIPES_MESSAGE } from '../../../../constants'

const props = {
  recipes,
  favoriteRecipe: recipes,
  userToggleRecipe: jest.fn(),
  handleClickRecipe: jest.fn(),
  userToggleVote: jest.fn(),
  userId: '1',
  type: 'primary',
}

describe('Components', () => {
  describe('RecipeList', () => {
    let component, wrapper

    beforeEach(() => {
      component = shallow(<RecipeList {...props} />)
      act(() => {
        wrapper = mount(<RecipeList {...props} />)
      })
    })

    it('Should render RecipeList component', () => {
      const renderComponent = renderer.create(<RecipeList {...props} />)
      expect(renderComponent).toMatchSnapshot()
    })

    it('Should render Loading component if is loading', () => {
      component.setProps({ loading: true })
      expect(component.find(Loading)).toHaveLength(1)
    })

    it('Should show no recipes message if have no recipes', () => {
      component.setProps({ recipes: [] })
      expect(
        component
          .find(Text)
          .at(2)
          .prop('children')
      ).toEqual(NO_RECIPES_MESSAGE)
    })

    it('Should show recipes list', () => {
      act(() => {
        wrapper
          .find(Text)
          .at(0)
          .props()
          .onPress()
      })

      act(() => {
        wrapper.update()
      })

      expect(wrapper.find(Recipe)).toHaveLength(props.recipes.length)

      // render with sort list
      act(() => {
        wrapper
          .find(Text)
          .at(1)
          .props()
          .onPress()
      })

      act(() => {
        wrapper.update()
      })

      expect(wrapper.find(Recipe)).toHaveLength(props.recipes.length)
    })

    it('Should call userToggleRecipe when toggle on recipe', () => {
      act(() => {
        wrapper
          .find(Text)
          .at(0)
          .props()
          .onPress()
      })

      act(() => {
        wrapper.update()
      })

      const recipe = wrapper.find(Recipe).at(0)
      recipe.props().handleSaveRecipe()
      expect(props.userToggleRecipe).toHaveBeenCalled()
    })

    it('Should call userToggleVote when toggle vote on recipe', () => {
      act(() => {
        wrapper
          .find(Text)
          .at(0)
          .props()
          .onPress()
      })

      act(() => {
        wrapper.update()
      })

      const recipe = wrapper.find(Recipe).at(0)
      recipe.props().handleToggleVote()
      expect(props.userToggleVote).toHaveBeenCalled()
    })
  })
})
