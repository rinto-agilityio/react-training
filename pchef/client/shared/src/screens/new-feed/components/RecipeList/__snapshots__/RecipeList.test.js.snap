// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components RecipeList Should render RecipeList component 1`] = `
<div
  className="css-view-1dbjc4n"
>
  <div
    className="css-view-1dbjc4n"
  >
    <div
      className="css-view-1dbjc4n r-borderBottomWidth-qklmqi r-borderColor-35fm6a r-display-6koalj r-flexDirection-18u37iz r-marginBottom-1yflyrw r-marginTop-1x0uki6"
    >
      <div
        className="css-text-901oao r-backgroundColor-1dw62tt r-borderTopLeftRadius-j9b53g r-borderTopRightRadius-1i97xy8 r-borderWidth-rs99b7 r-color-jwli3a r-cursor-1loqt21 r-paddingBottom-1inuy60 r-paddingLeft-b5h31w r-paddingRight-1ah4tor r-paddingTop-glunga r-textTransform-tsynxw"
        data-focusable={true}
        dir="auto"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
      >
         All Recipes
      </div>
      <div
        className="css-text-901oao r-backgroundColor-1pzepah r-borderTopLeftRadius-j9b53g r-borderTopRightRadius-1i97xy8 r-borderWidth-rs99b7 r-color-jwli3a r-cursor-1loqt21 r-paddingBottom-1inuy60 r-paddingLeft-b5h31w r-paddingRight-1ah4tor r-paddingTop-glunga r-textTransform-tsynxw"
        data-focusable={true}
        dir="auto"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex="0"
      >
         Top Votest Recipes
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
    >
      <div
        className="css-view-1dbjc4n r-display-6koalj"
      >
        <div
          className="css-view-1dbjc4n"
        >
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-color-cqee49 r-cursor-1loqt21 r-fontSize-adyw6z r-fontWeight-1kfrs79 r-maxWidth-14qjzug r-overflow-1udh08x r-paddingLeft-b5h31w r-paddingRight-1ah4tor"
            data-focusable={true}
            dir="auto"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex="0"
          >
            Bun Bo Hue Recipe
          </div>
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-color-1h2fve2 r-fontSize-1i10wst r-fontWeight-1kfrs79 r-height-1g7fiml r-maxWidth-14qjzug r-overflow-1udh08x r-paddingLeft-b5h31w r-paddingRight-1ah4tor"
            dir="auto"
          >
            2lb pork neck bones, 2lb beef bones, 1lb oxtail
          </div>
          <div
            className="css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
            data-focusable={true}
            disabled={false}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            tabIndex="0"
          >
            <div
              className="css-view-1dbjc4n r-backgroundColor-7klgxt r-height-bajy9j r-marginTop-156q2ks r-width-13qz1uu"
            >
              <div
                className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-height-1pi2tsx r-overflow-1udh08x r-width-13qz1uu r-zIndex-417010"
              >
                <div
                  className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
                  style={
                    Object {
                      "backgroundImage": "url(\\"https://www.tastefullysimple.com/_/media/images/recipe-default-image.png\\")",
                    }
                  }
                />
                <img
                  alt=""
                  className="css-accessibilityImage-9pa8cd"
                  draggable={false}
                  src="https://www.tastefullysimple.com/_/media/images/recipe-default-image.png"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-1wtj0ep r-padding-1pcd2l5"
        >
          <div
            className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
          >
            <div
              className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
              data-focusable={true}
              disabled={false}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="css-hitSlop-mjp8i1"
                style={
                  Object {
                    "bottom": "-6px",
                    "left": "-6px",
                    "right": "-6px",
                    "top": "-6px",
                  }
                }
              />
              <div
                className="css-view-1dbjc4n"
              >
                <div
                  className="css-text-901oao r-backgroundColor-1niwhzg r-pointerEvents-633pao"
                  dir="auto"
                  focusable={false}
                  role="img"
                  style={
                    Object {
                      "WebkitTransform": "scaleX(1)",
                      "color": "rgba(211,43,83,1.00)",
                      "fontFamily": "MaterialIcons",
                      "fontSize": "30px",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "transform": "scaleX(1)",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <div
              className="css-text-901oao r-marginLeft-1n0xq6e r-textAlign-q4m81j"
              dir="auto"
            >
              Save
            </div>
          </div>
          <div
            className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
          >
            <div
              className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
              data-focusable={true}
              disabled={false}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="css-hitSlop-mjp8i1"
                style={
                  Object {
                    "bottom": "-6px",
                    "left": "-6px",
                    "right": "-6px",
                    "top": "-6px",
                  }
                }
              />
              <div
                className="css-view-1dbjc4n"
              >
                <div
                  className="css-text-901oao r-backgroundColor-1niwhzg r-pointerEvents-633pao"
                  dir="auto"
                  focusable={false}
                  role="img"
                  style={
                    Object {
                      "WebkitTransform": "scaleX(1)",
                      "color": "rgba(70,91,174,1.00)",
                      "fontFamily": "MaterialIcons",
                      "fontSize": "30px",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "transform": "scaleX(1)",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <div
              className="css-text-901oao r-marginLeft-1n0xq6e r-textAlign-q4m81j"
              dir="auto"
            >
              2
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-view-1dbjc4n r-display-6koalj"
      >
        <div
          className="css-view-1dbjc4n"
        >
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-color-cqee49 r-cursor-1loqt21 r-fontSize-adyw6z r-fontWeight-1kfrs79 r-maxWidth-14qjzug r-overflow-1udh08x r-paddingLeft-b5h31w r-paddingRight-1ah4tor"
            data-focusable={true}
            dir="auto"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex="0"
          >
            Crispy Fried Chicken
          </div>
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-color-1h2fve2 r-fontSize-1i10wst r-fontWeight-1kfrs79 r-height-1g7fiml r-maxWidth-14qjzug r-overflow-1udh08x r-paddingLeft-b5h31w r-paddingRight-1ah4tor"
            dir="auto"
          >
            Description for Crispy Fried Chicken
          </div>
          <div
            className="css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
            data-focusable={true}
            disabled={false}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            tabIndex="0"
          >
            <div
              className="css-view-1dbjc4n r-backgroundColor-7klgxt r-height-bajy9j r-marginTop-156q2ks r-width-13qz1uu"
            >
              <div
                className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-height-1pi2tsx r-overflow-1udh08x r-width-13qz1uu r-zIndex-417010"
              >
                <div
                  className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
                  style={
                    Object {
                      "backgroundImage": "url(\\"https://www.tastefullysimple.com/_/media/images/recipe-default-image.png\\")",
                    }
                  }
                />
                <img
                  alt=""
                  className="css-accessibilityImage-9pa8cd"
                  draggable={false}
                  src="https://www.tastefullysimple.com/_/media/images/recipe-default-image.png"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-1wtj0ep r-padding-1pcd2l5"
        >
          <div
            className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
          >
            <div
              className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
              data-focusable={true}
              disabled={false}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="css-hitSlop-mjp8i1"
                style={
                  Object {
                    "bottom": "-6px",
                    "left": "-6px",
                    "right": "-6px",
                    "top": "-6px",
                  }
                }
              />
              <div
                className="css-view-1dbjc4n"
              >
                <div
                  className="css-text-901oao r-backgroundColor-1niwhzg r-pointerEvents-633pao"
                  dir="auto"
                  focusable={false}
                  role="img"
                  style={
                    Object {
                      "WebkitTransform": "scaleX(1)",
                      "color": "rgba(211,43,83,1.00)",
                      "fontFamily": "MaterialIcons",
                      "fontSize": "30px",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "transform": "scaleX(1)",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <div
              className="css-text-901oao r-marginLeft-1n0xq6e r-textAlign-q4m81j"
              dir="auto"
            >
              Save
            </div>
          </div>
          <div
            className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
          >
            <div
              className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
              data-focusable={true}
              disabled={false}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="css-hitSlop-mjp8i1"
                style={
                  Object {
                    "bottom": "-6px",
                    "left": "-6px",
                    "right": "-6px",
                    "top": "-6px",
                  }
                }
              />
              <div
                className="css-view-1dbjc4n"
              >
                <div
                  className="css-text-901oao r-backgroundColor-1niwhzg r-pointerEvents-633pao"
                  dir="auto"
                  focusable={false}
                  role="img"
                  style={
                    Object {
                      "WebkitTransform": "scaleX(1)",
                      "color": "rgba(70,91,174,1.00)",
                      "fontFamily": "MaterialIcons",
                      "fontSize": "30px",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "transform": "scaleX(1)",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <div
              className="css-text-901oao r-marginLeft-1n0xq6e r-textAlign-q4m81j"
              dir="auto"
            >
              2
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-view-1dbjc4n r-display-6koalj"
      >
        <div
          className="css-view-1dbjc4n"
        >
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-color-cqee49 r-cursor-1loqt21 r-fontSize-adyw6z r-fontWeight-1kfrs79 r-maxWidth-14qjzug r-overflow-1udh08x r-paddingLeft-b5h31w r-paddingRight-1ah4tor"
            data-focusable={true}
            dir="auto"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex="0"
          >
            Crispy Fried Chicken
          </div>
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-color-1h2fve2 r-fontSize-1i10wst r-fontWeight-1kfrs79 r-height-1g7fiml r-maxWidth-14qjzug r-overflow-1udh08x r-paddingLeft-b5h31w r-paddingRight-1ah4tor"
            dir="auto"
          >
            Description for Crispy Fried Chicken
          </div>
          <div
            className="css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
            data-focusable={true}
            disabled={false}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            tabIndex="0"
          >
            <div
              className="css-view-1dbjc4n r-backgroundColor-7klgxt r-height-bajy9j r-marginTop-156q2ks r-width-13qz1uu"
            >
              <div
                className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-height-1pi2tsx r-overflow-1udh08x r-width-13qz1uu r-zIndex-417010"
              >
                <div
                  className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
                  style={
                    Object {
                      "backgroundImage": "url(\\"https://www.tastefullysimple.com/_/media/images/recipe-default-image.png\\")",
                    }
                  }
                />
                <img
                  alt=""
                  className="css-accessibilityImage-9pa8cd"
                  draggable={false}
                  src="https://www.tastefullysimple.com/_/media/images/recipe-default-image.png"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-1wtj0ep r-padding-1pcd2l5"
        >
          <div
            className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
          >
            <div
              className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
              data-focusable={true}
              disabled={false}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="css-hitSlop-mjp8i1"
                style={
                  Object {
                    "bottom": "-6px",
                    "left": "-6px",
                    "right": "-6px",
                    "top": "-6px",
                  }
                }
              />
              <div
                className="css-view-1dbjc4n"
              >
                <div
                  className="css-text-901oao r-backgroundColor-1niwhzg r-pointerEvents-633pao"
                  dir="auto"
                  focusable={false}
                  role="img"
                  style={
                    Object {
                      "WebkitTransform": "scaleX(1)",
                      "color": "rgba(211,43,83,1.00)",
                      "fontFamily": "MaterialIcons",
                      "fontSize": "30px",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "transform": "scaleX(1)",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <div
              className="css-text-901oao r-marginLeft-1n0xq6e r-textAlign-q4m81j"
              dir="auto"
            >
              Save
            </div>
          </div>
          <div
            className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
          >
            <div
              className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
              data-focusable={true}
              disabled={false}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="css-hitSlop-mjp8i1"
                style={
                  Object {
                    "bottom": "-6px",
                    "left": "-6px",
                    "right": "-6px",
                    "top": "-6px",
                  }
                }
              />
              <div
                className="css-view-1dbjc4n"
              >
                <div
                  className="css-text-901oao r-backgroundColor-1niwhzg r-pointerEvents-633pao"
                  dir="auto"
                  focusable={false}
                  role="img"
                  style={
                    Object {
                      "WebkitTransform": "scaleX(1)",
                      "color": "rgba(70,91,174,1.00)",
                      "fontFamily": "MaterialIcons",
                      "fontSize": "30px",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "transform": "scaleX(1)",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <div
              className="css-text-901oao r-marginLeft-1n0xq6e r-textAlign-q4m81j"
              dir="auto"
            >
              2
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-view-1dbjc4n r-display-6koalj"
      >
        <div
          className="css-view-1dbjc4n"
        >
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-color-cqee49 r-cursor-1loqt21 r-fontSize-adyw6z r-fontWeight-1kfrs79 r-maxWidth-14qjzug r-overflow-1udh08x r-paddingLeft-b5h31w r-paddingRight-1ah4tor"
            data-focusable={true}
            dir="auto"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex="0"
          >
            Crispy Fried Chicken
          </div>
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-color-1h2fve2 r-fontSize-1i10wst r-fontWeight-1kfrs79 r-height-1g7fiml r-maxWidth-14qjzug r-overflow-1udh08x r-paddingLeft-b5h31w r-paddingRight-1ah4tor"
            dir="auto"
          >
            Description for Crispy Fried Chicken
          </div>
          <div
            className="css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
            data-focusable={true}
            disabled={false}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            tabIndex="0"
          >
            <div
              className="css-view-1dbjc4n r-backgroundColor-7klgxt r-height-bajy9j r-marginTop-156q2ks r-width-13qz1uu"
            >
              <div
                className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-height-1pi2tsx r-overflow-1udh08x r-width-13qz1uu r-zIndex-417010"
              >
                <div
                  className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
                  style={
                    Object {
                      "backgroundImage": "url(\\"https://www.tastefullysimple.com/_/media/images/recipe-default-image.png\\")",
                    }
                  }
                />
                <img
                  alt=""
                  className="css-accessibilityImage-9pa8cd"
                  draggable={false}
                  src="https://www.tastefullysimple.com/_/media/images/recipe-default-image.png"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-view-1dbjc4n r-display-6koalj r-flexDirection-18u37iz r-justifyContent-1wtj0ep r-padding-1pcd2l5"
        >
          <div
            className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
          >
            <div
              className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
              data-focusable={true}
              disabled={false}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="css-hitSlop-mjp8i1"
                style={
                  Object {
                    "bottom": "-6px",
                    "left": "-6px",
                    "right": "-6px",
                    "top": "-6px",
                  }
                }
              />
              <div
                className="css-view-1dbjc4n"
              >
                <div
                  className="css-text-901oao r-backgroundColor-1niwhzg r-pointerEvents-633pao"
                  dir="auto"
                  focusable={false}
                  role="img"
                  style={
                    Object {
                      "WebkitTransform": "scaleX(1)",
                      "color": "rgba(211,43,83,1.00)",
                      "fontFamily": "MaterialIcons",
                      "fontSize": "30px",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "transform": "scaleX(1)",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <div
              className="css-text-901oao r-marginLeft-1n0xq6e r-textAlign-q4m81j"
              dir="auto"
            >
              Save
            </div>
          </div>
          <div
            className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
          >
            <div
              className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-margin-5scogr r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
              data-focusable={true}
              disabled={false}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="css-hitSlop-mjp8i1"
                style={
                  Object {
                    "bottom": "-6px",
                    "left": "-6px",
                    "right": "-6px",
                    "top": "-6px",
                  }
                }
              />
              <div
                className="css-view-1dbjc4n"
              >
                <div
                  className="css-text-901oao r-backgroundColor-1niwhzg r-pointerEvents-633pao"
                  dir="auto"
                  focusable={false}
                  role="img"
                  style={
                    Object {
                      "WebkitTransform": "scaleX(1)",
                      "color": "rgba(70,91,174,1.00)",
                      "fontFamily": "MaterialIcons",
                      "fontSize": "30px",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "transform": "scaleX(1)",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
            <div
              className="css-text-901oao r-marginLeft-1n0xq6e r-textAlign-q4m81j"
              dir="auto"
            >
              2
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
